### PR TITLE
[jmx] fix crash when jmx_max_restarts < 1

### PR DESCRIPTION
**Pull request by vickenty**
### What does this PR do?

Fix a crash in the agent when jmxfetch is enabled and `jmx_max_restarts` setting is set to value less than 1.

### Describe your test plan

Enable JMX collection (`cp bin/agent/dist/conf.d/jmx.d/conf.yaml{.example,}`) and set `jmx_max_restarts: 0` in `datadog.yaml`

### CODE REVIEW

File: pkg/jmxfetch/limiter.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package jmxfetch
+
+import "time"
+
+// restartLimiter checks if too many restarts happened within a time period
+type restartLimiter struct {
+	maxRestarts int
+	interval    float64
```
Comment (gh123man) [inline pkg/jmxfetch/limiter.go:8]: Could you add a comment on what unit the interval is? (seconds, etc)

File: pkg/jmxfetch/limiter.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	}
+}
+
+func (r *restartLimiter) canRestart(now time.Time) bool {
+	if r.maxRestarts < 1 {
+		return false
+	}
+
+	r.stopTimes[r.idx] = now
+	oldestIdx := (r.idx + 1) % r.maxRestarts
```
Comment (gh123man) [inline pkg/jmxfetch/limiter.go:32]: This used to be `oldestIdx := (idx + maxRestarts + 1) % maxRestarts` - did you intend to drop `maxRestarts` from the calculation?

File: pkg/jmxfetch/limiter_test.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	}
+
+	tref := time.Now()
+	for tidx, tt := range tests {
+		r := newRestartLimiter(tt.maxRestarts, tt.interval)
+		for cidx, check := range tt.checks {
+			name := fmt.Sprintf("%d/%d", tidx, cidx)
+			t.Run(name, func(t *testing.T) {
+				if got := r.canRestart(tref.Add(check.now)); got != check.result {
+					t.Errorf("restartLimiter.canRestart() = %v, want %v", got, check.result)
```
Comment (ogaca-dd) [inline pkg/jmxfetch/limiter_test.go:34]: You can also use `require.Equal(t, got, check.result)`.

**Review by ogaca-dd — APPROVED** [review]
LGTM.
You should either add a release note or add the label `changelog/no-changelog`.

File: pkg/jmxfetch/limiter.go (hunk 2 above)
Comment (vickenty) [reply pkg/jmxfetch/limiter.go:32]: Yes, as far as I can tell, adding `maxRestarts` to the lhs is redundant in this case (all values are non-negative here). Do you think we should keep it anyway?

Comment (gh123man) [reply pkg/jmxfetch/limiter.go:32]: Makes sense - just wanted to double check. 
in that case line `39` below can just be: `r.idx = oldestIdx` so you don't have to do it twice :)

**Review by gh123man — APPROVED** [review]

**Pull request merged**